lib/aur-fetch

```diff
@@ -1,67 +1,58 @@
 #!/bin/bash
 readonly argv0=fetch
 readonly PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
-readonly aurweb='https://aur.archlinux.org'
-readonly urlpath='/cgit/aur.git/snapshot'
 set -o errexit
 
-declare    logdir=/dev/null
-declare -i snapshot=0 default_path=1
+declare logdir=/dev/null
+declare mode=git
 
 clone() {
-    declare -r remote=$1 logdir=$2
+    declare -r log_dir=$1
 
     while IFS= read -r; do
+        declare git_dir=${REPLY##*/}     # strip path
+        declare git_dir=${git_dir%%.git} # strip .git suffix
+
         # Avoid issues with exotic file system layouts (#274)
-        git_wrap() { git --work-tree="$REPLY" --git-dir="$REPLY"/.git "$@"; }
+        git_wrap() { git --work-tree="$git_dir" --git-dir="$git_dir"/.git "$@"; }
         
-        if [[ -d $REPLY/.git ]]; then
+        if [[ -d $git_dir/.git ]]; then
             # returns 0 even if nothing was fetched
             git_wrap fetch -v
 
             if [[ $(git_wrap rev-parse HEAD) != $(git_wrap rev-parse '@{upstream}') ]]; then
-                if [[ -d $logdir ]]; then
+                if [[ -d $log_dir ]]; then
                     git_wrap --no-pager log --patch --stat '..@{upstream}' \
-                        >> "$logdir/$REPLY".patch
+                        >> "$log_dir/$git_dir".patch
                 fi
 
                 git_wrap reset --hard HEAD
                 git_wrap merge
             fi
         else
-            git clone "$remote/$REPLY".git
+            git clone "$REPLY"
         fi
     done
 }
 readonly -f clone
 
 usage() {
-    printf '%s\n' "usage: $argv0 [-L logdir] [-t]" >&2
+    printf '%s\n' "usage: $argv0 [-L log_dir] [-t]" >&2
```

Comment:
`printf '%s\n' "usage: $argv0 [-L log_dir] [-t mode]" >&2`

Reply:
The idea was that the default "mode" is `git`, which is then set to `tar` if you specify the `-t` flag. Should be less typing and matches the old `aur-fetch` interface. I guess the initial mode declaration can be moved from the top of the script to getopt to make this more clear.

Reply:
Right, this makes sense. I'll update the documentation accordingly